x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/Expressions.java

```diff
@@ -102,6 +103,31 @@ public static AttributeSet references(List<? extends Expression> exps) {
         return set;
     }
 
+    public static AttributeSet referencesExcept(Set<? extends Expression> exps, AttributeSet except) {
```

Comment:
Maybe a name with a verb, something `filterReferences` ?

Reply:
I would name it `excluded`.

Reply:
ok